packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,351 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IResource, Names, Resource, Tag, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage, TagOption } from './common';
+import { EventNotificationsProps, LaunchRoleProps, ProvisioningRulesProps, TagUpdatesProps, StackSetConstraintProps } from './constraints';
+import { AssociationManager } from './private/association-manager';
+import { getIdentifier } from './private/util';
+import { InputValidator } from './private/validation';
+import { IProduct } from './product';
+import { CfnPortfolio, CfnPortfolioPrincipalAssociation, CfnPortfolioShare } from './servicecatalog.generated';
+
+/**
+ * A Service Catalog portfolio.
+ */
+export interface IPortfolio extends IResource {
+
+  /**
+   * The ARN of the portfolio.
+   * @attribute
+   */
+  readonly portfolioArn: string;
+
+  /**
+   * The ID of the portfolio.
+   * @attribute
+   */
+  readonly portfolioId: string;
+
+  /**
+   * The name of the portfolio.
+   * @attribute
+   */
+  readonly portfolioName: string;
+
+  /**
+   * Associate portfolio with an IAM Role.
+   * @param role an IAM role
+   */
+  giveAccessToRole(role: iam.IRole): void;
+
+  /**
+   * Associate portfolio with an IAM User.
+   * @param user an IAM user
+   */
+  giveAccessToUser(user: iam.IUser): void;
+
+  /**
+   * Associate portfolio with a principal
+   * (Group).
+   * @param group an IAM Group
+   */
+  giveAccessToGroup(group: iam.IGroup): void;
+
+  /**
+   * Associate portfolio with the given product.
+   * @param product A service catalog produt.
+   */
+  addProduct(product: IProduct): void;
+
+  /**
+   * Share portfolio with another account.
+   * @param accountId AWS account to share portfolio with
+   * @param shareTagOptions ability to share or not share tag options
+   */
+  share(accountId: string, shareTagOptions?: boolean, acceptLanguage?: AcceptLanguage): void;
```

Comment:
Let's bundle the 2 optional arguments into an interface:

```suggestion
export interface PortfolioShareOptions {
  readonly shareTagOptions?: boolean;
  readonly acceptLanguage?: AcceptLanguage;
}

// ...
  share(accountId: string , options: PortfolioShareOptions): void;
```

Otherwise, if you want to pass `acceptLanguage`, but not `shareTagOptions`, you have to call it `share(account, undefined, AcceptLanguage.EN)`, which is awkward.

Reply:
good call